### ISSUE TITLE
triage issues assigned to samreid

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/tasks#234]
(5/7/2015 developer meeting)

Triage issues assigned to @samreid, to see if any of these should be reassigned while he's on leave. @ariel-phet will handle.

**Comment by samreid** [comment]
One issue we discussed at a PhET developer meeting a while ago that was assigned to me but never completed was:

> Let’s move phetbrand/brand back to phetsims/phetbrand. And make it private.  And an adaptedfromphet repo.  SR will do this.

There is an issue for "adapted from" features here: https://github.com/phetsims/brand/issues/2 But as far as I know, no issues for moving the brand repo and making it private.

**Comment by samreid** [comment]
@kathy-phet asked me to look into buggy tab key in photoelectric effect 10 months ago, and I spent 2.5 hours on it.  The issue:

Fix Photoelectric Effect tabbing: https://phet.unfuddle.com/a#/projects/9404/tickets/by_number/3633 see photoelectric#3281

My report:

> I have spent about 2.5 hours looking into this. It is very convoluted and I am unfamiliar with the physics model and implementation. This far in, I do not see any simple fixes. The issues appear with how different controls set values and update each other, and there are numerous different conversions that must be made forward and backwards, and multiple modes (whether the slider is in intensity mode or rate mode), and the fact that intensity is a function of wavelength AND maximum rate. Controls change each other's values, which trigger further changes to the controls. This could take another 6+ hours to resolve and will probably require thorough testing.

@kathy-phet response:

> Given you time estimate - I think we need to put this one on the shelf until late August --- after Ido's study and after ODF sim deadlines.

**Comment by samreid** [comment]
It would be good to discuss with @emily-phet if anything else will be required for accessibility in the near future.

**Comment by samreid** [comment]
Oh one more issue I just remembered.  We will probably get a new version of CCK's NoiseGenerator.java soon and someone may need to update CCK code with the new file, and republish a dev version of CCK.

**Comment by emily-phet** [comment]
@samreid - you mentioned last week that you wanted to meet to discuss the Paciello group meeting. Could you summarize if there are any changes to implementation approaches you would recommend based on that meeting?

**Comment by samreid** [comment]
@jonathanolson & I discussed accessibility implementation details briefly after the Paciello group meeting and agreed that *based on Paciello group answers to technical questions* the "parallel DOM tree" strategy would be a viable alternative to the "canvas subdoms + svg aria markup" strategy which we were previously considering.  This basically frees us up to use an approach that may be (a) easier to implement, (b) independent of renderer used (webgl vs svg vs canvas), (c) would work on more browsers out of the box.  We previously were avoiding the "parallel DOM approach" because it is somewhat unconventional, but the Paciello group could not think of any a priori technical reasons that it would not work--as long as we provide the appropriate dom elements and test that it works with a number of assistive technologies on different platforms.

@jonathanolson can you chime in and provide any additional details I've omitted?

**Comment by emily-phet** [comment]
Thanks @samreid - that's useful to know. Perhaps @jonathanolson can weigh in on his thoughts from the meeting, and also a brief description of the basic differences between the subDOM vs parallel DOM approaches.
I believe I have a general idea, but it would be useful to lay out some basic structural differences and the impacts of each (e.g., how each implementation may effect tablets, phones, screen reader compatibility, etc.).

**Comment by jonathanolson** [comment]
Normal sub-DOM like structure:
```
SVG stuff on top (SVG-specific accessibility features for stuff displayed here is added to the visual elements)
Canvas stuff on the bottom (sub-DOM peers are put under the Canvas element, and are invisible. totally different than SVG)
```
With this approach, some features may be supported with SVG in some browsers, some may be supported with Canvas in browsers, and we have to do at least double the work at implementing things. Has many problems when one "thing" you want selected happens to be displayed by multiple layers (SVG and Canvas).

A "parallel" DOM tree would allow us to not use mixed SVG/Canvas-sub-DOM ways of specifying accessibility (which has many pitfalls), and instead have a simplified structure:
```
SVG stuff on top
Canvas stuff on bottom
Invisible DOM tree that handles all accessibility concerns
```
This allows us to support keyboard accessibility nicely on all browsers (for free, basically), and do a much simpler implementation (probably a quarter of the development time). No problems with "things" that are displayed across multiple layers, as our "display" is separate from our "accessibility". No parallel implementations.

I highly prefer the "one DOM tree for accessibility" now that the group approved of the approach.

**Comment by samreid** [comment]
@jonathanolson one possibility I was considering for accessibility: since we are moving toward the "One parallel DOM tree for accessibility" approach, it seems there may be some advantages in powering it using the together.js engine.  I created an issue in the together repository where we can discuss it, please see https://github.com/phetsims/together/issues/118

**Comment by jonathanolson** [comment]
Responded. Hopefully Scenery can provide a good accessibility framework and implementation, that could potentially be hooked into with together.

**Comment by ariel-phet** [comment]
This issue seems to be stale now that @samreid  is ramping hours up.
